app/models/request/sample_compound_aliquot_transfer.rb

```diff
@@ -10,25 +10,53 @@ module Request::SampleCompoundAliquotTransfer
   # at the source has a different tag_depth defined
   def compound_samples_needed?
     return false if asset.aliquots.count == 1
-    (asset.aliquots.pluck(:tag_depth).uniq.count == asset.aliquots.count)
+
+    _tag_clash?
   end
 
   # Creates a sample in a single aliquot at destination as a compound sample
   # where the component samples are all samples at source
-  def transfer_aliquots_into_compound_sample_aliquot
-    compound_sample = _create_compound_sample(_default_compound_study, asset.samples)
+  def transfer_aliquots_into_compound_sample_aliquots
+    _aliquots_by_tags_combination.each do |_tags_combo, aliquot_list|
+      transfer_into_compound_sample_aliquot(aliquot_list)
+    end
+  end
+
+  private
+
+  def transfer_into_compound_sample_aliquot(source_aliquots)
+    samples = source_aliquots.map(&:sample)
+
+    if source_aliquots.pluck(:tag_depth).uniq.count != source_aliquots.size
```

Comment:
This could be renamed the tag_clash

Reply:
added a comment to explain what I'm checking for